internal/providers/aws/aws.go

```diff
@@ -18,30 +18,19 @@
 package aws
 
 import (
-	"net/url"
-
-	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
-	"github.com/coreos/ignition/v2/internal/log"
-	"github.com/coreos/ignition/v2/internal/providers/util"
-	"github.com/coreos/ignition/v2/internal/resource"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
 	"github.com/aws/aws-sdk-go/aws/ec2metadata"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/coreos/ignition/v2/config/v3_2_experimental/types"
+	"github.com/coreos/ignition/v2/internal/log"
+	"github.com/coreos/ignition/v2/internal/providers/util"
+	"github.com/coreos/ignition/v2/internal/resource"
 	"github.com/coreos/vcontext/report"
 )
 
-var (
-	userdataUrl = url.URL{
-		Scheme: "http",
-		Host:   "169.254.169.254",
-		Path:   "2019-10-01/user-data",
-	}
-)
-
 func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
-	data, err := f.FetchToBuffer(userdataUrl, resource.FetchOptions{})
+	data, err := ec2metadata.New(session.Must(session.NewSession())).GetUserData()
```

Comment:
@jlebon would this need an explicit check to return `ErrNeedNet` if we're running in offline mode?

Right now we're using `FetchToBuffer` which has [this block of code](https://github.com/coreos/ignition/blob/master/internal/resource/url.go#L121-L123) which will return `ErrNeedNet` if Ignition is in offline mode (as we're attempting to fetch a http url).